ironsftp.1

```diff
@@ -0,0 +1,675 @@
+.\" THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+.\" (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+.\" THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+.\"
+.Dd Aug. 25 2016
+.Dt IRONSFTP 1
+.Os
+.Sh NAME
+.Nm ironsftp
+.Nd end-to-end secure file transfer program
+.Sh SYNOPSIS
+.Nm ironsftp
+.Bk -words
+.Op Fl 1246aCfpqrv
+.Op Fl B Ar buffer_size
+.Op Fl b Ar batchfile
+.Op Fl c Ar cipher
+.Op Fl D Ar sftp_server_path
+.Op Fl F Ar ssh_config
+.Op Fl i Ar identity_file
+.Op Fl l Ar limit
+.Op Fl o Ar ssh_option
+.Op Fl P Ar port
+.Op Fl R Ar num_requests
+.Op Fl S Ar program
+.Op Fl s Ar subsystem | sftp_server
+.Ar host
+.Ek
+.Nm ironsftp
+.Oo Ar user Ns @ Oc Ns
+.Ar host Ns Op : Ns Ar
+.Nm ironsftp
+.Oo
+.Ar user Ns @ Oc Ns
+.Ar host Ns Oo : Ns Ar dir Ns
+.Op Ar /
+.Oc
+.Nm ironsftp
+.Fl b Ar batchfile
+.Oo Ar user Ns @ Oc Ns Ar host
+.Sh DESCRIPTION
+.Nm
+is an "end-to-end secure" interactive file transfer program, based on the
+utility 
+.Xr sftp 1 .
+Like sftp,
+.Nm
+performs all operations over an encrypted
+.Xr ssh 1
+transport. However, it protects the files after they are written on the server.
+When you write a file to the server (using 'put'), the file is automatically
+encrypted before it is uploaded. When you fetch a file from the server
+(using 'get'), the file is automatically decrypted after it is downloaded.
+These secured files have a '.iron' suffix added to indicate that they have
+been encrypted.
+.Pp
+Files can also be securely shared with other users. Before you upload a file,
+you can specify that additional "recipients" should be able to download and
+decrypt the file as well. As long as
+.Nm
+can retrieve those users' public key information, the file will be encrypted
+such that any of those users can also download and decrypt the file. Anyone
+that is not a registered recipient might be able to retrieve the file,
+depending on file permissions, but will not be able to decrypt it to access
+the original contents.
+.Pp
+Typically, 
+.Nm
+connects and logs into the specified
+.Ar host ,
+then enters an interactive command mode.
+.Pp
+The second usage format will retrieve files automatically if a non-interactive
+authentication method is used; otherwise it will do so after
+successful interactive authentication.
+.Pp
+The third usage format allows
+.Nm
+to start in a remote directory.
+.Pp
+The final usage format allows for automated sessions using the
+.Fl b
+option.
+In such cases, it is necessary to configure non-interactive authentication
+to obviate the need to enter a password at connection time (see
+.Xr sshd 8
+and
+.Xr ssh-keygen 1
+for details).
+.Pp
+Since some usage formats use colon characters to delimit host names from path
+names, IPv6 addresses must be enclosed in square brackets to avoid ambiguity.
+.Pp
+The options are as follows:
+.Bl -tag -width Ds
+.It Fl 1
+Specify the use of protocol version 1.
```

Comment:
Gosh. This just reminds me that sshv1 is still around. Really tempting to strip that out of our stuff.